code/game/objects/items/devices/radio/radio.dm

```diff
@@ -397,7 +397,9 @@
 			// left hands are odd slots
 			if (idx && (idx % 2) == (message_mods[RADIO_EXTENSION] == MODE_L_HAND))
 				return
-
+	//prevents soapbox speaking from leaking into hotmics
+	if(message_mods[SAY_FAKE_COMMAND])
+		spans -= SPAN_COMMAND
```

Comment:
You don't need to check in radio code for this, you should just not add the span if the mob is speaking over the radio.

Reply:
The reason for this one is for when a radio is set to open mic, It already doesn't add the span to the radio if they use a radio normally, but if its just listening and picking everything up around it I couldn't figure that out, if you have any advice on that, otherwise I'll ask around

Reply:
I see

I'm believe this is solvable in a cleaner way but you can leave it as is, I don't remember off the top of my head what that way is